src/CADL.Extension/Emitter.Csharp/src/lib/operation.ts

```diff
@@ -0,0 +1,446 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { getOperationLink } from "@azure-tools/cadl-azure-core";
+import {
+    createDpgContext,
+    isApiVersion,
+    shouldGenerateConvenient,
+    shouldGenerateProtocol
+} from "@azure-tools/cadl-dpg";
+import {
+    EmitContext,
+    getDeprecated,
+    getDoc,
+    getSummary,
+    isErrorModel,
+    Model,
+    ModelProperty,
+    Namespace,
+    Operation,
+    Program
+} from "@cadl-lang/compiler";
+import { getResourceOperation, ResourceOperation } from "@cadl-lang/rest";
+import {
+    HttpOperation,
+    HttpOperationParameter,
+    HttpOperationResponse
+} from "@cadl-lang/rest/http";
+import { NetEmitterOptions } from "../options.js";
+import { BodyMediaType, typeToBodyMediaType } from "../type/BodyMediaType.js";
+import { collectionFormatToDelimMap } from "../type/CollectionFormat.js";
+import { HttpResponseHeader } from "../type/HttpResponseHeader.js";
+import { InputConstant } from "../type/InputConstant.js";
+import { InputOperation } from "../type/InputOperation.js";
+import { InputOperationParameterKind } from "../type/InputOperationParameterKind.js";
+import { InputParameter } from "../type/InputParameter.js";
+import {
+    InputEnumType,
+    InputListType,
+    InputModelType,
+    InputType,
+    isInputLiteralType,
+    isInputUnionType
+} from "../type/InputType.js";
+import { OperationFinalStateVia } from "../type/OperationFinalStateVia.js";
+import { OperationLongRunning } from "../type/OperationLongRunning.js";
+import { OperationPaging } from "../type/OperationPaging.js";
+import { OperationResponse } from "../type/OperationResponse.js";
+import {
+    RequestLocation,
+    requestLocationMap
+} from "../type/RequestLocation.js";
+import {
+    parseHttpRequestMethod,
+    RequestMethod
+} from "../type/RequestMethod.js";
+import { getExternalDocs, getOperationId, hasDecorator } from "./decorators.js";
+import {
+    getDefaultValue,
+    getEffectiveSchemaType,
+    getInputType
+} from "./model.js";
+import { capitalize } from "./utils.js";
+
+export function loadOperation(
+    context: EmitContext<NetEmitterOptions>,
+    operation: HttpOperation,
+    uri: string,
+    urlParameters: InputParameter[] | undefined = undefined,
+    serviceNamespaceType: Namespace,
+    models: Map<string, InputModelType>,
+    enums: Map<string, InputEnumType>
```

Comment:
I saw a lot of functions are taking these two maps, shall we refactor the whole thing into a class and put these things as properties or fields of that class, and fucntions like this could be instance method and it does not have to take these maps as parameters, they will be internal state of the `builder` if you will

Reply:
@ArcturusZhang `models` and `enums` are golbal variables for the whole codeModel. It can be only be the fields of the codeModel.  And then every function such as loadOperation, getInputModel will be the function of builder. There is only one class, one file, it is not good, either. 
or, we can split a lot of builder, operationBuilder, modelBuilder, and those two will be set in their constructor. But now they are parameters of constuctor instead of function.

or, we will define an global varaible for these, and it will be imported in any ts files. this approach I don't think it is good for encapsulation.

We can discuss this, and maybe delay to implement it.